src/JSInterop/Microsoft.JSInterop.JS/src/src/Microsoft.JSInterop.ts

```diff
@@ -394,24 +411,28 @@ export module DotNet {
   }
 
   const dotNetObjectRefKey = '__dotNetObject';
-  attachReviver(function reviveDotNetObject(key: any, value: any) {
-    if (value && typeof value === 'object' && value.hasOwnProperty(dotNetObjectRefKey)) {
-      return new DotNetObject(value.__dotNetObject);
-    }
-
-    // Unrecognized - let another reviver handle it
-    return value;
-  });
-
-  attachReviver(function reviveJSObjectReference(key: any, value: any) {
-    if (value && typeof value === 'object' && value.hasOwnProperty(jsObjectIdKey)) {
-      const id = value[jsObjectIdKey];
-      const jsObject = cachedJSObjectsById[id];
+  const byteArrayRefKey = '__byte[]';
+  attachReviver(function reviveReference(key: any, value: any) {
+    if (value && typeof value === 'object') {
+      if (value.hasOwnProperty(dotNetObjectRefKey)) {
+        return new DotNetObject(value.__dotNetObject);
+      } else if (value.hasOwnProperty(jsObjectIdKey)) {
+        const id = value[jsObjectIdKey];
+        const jsObject = cachedJSObjectsById[id];
+
+        if (jsObject) {
+          return jsObject.getWrappedObject();
+        } else {
+          throw new Error(`JS object instance with ID ${id} does not exist (has it been disposed?).`);
```

Comment:
We probably don't want to be asking questions in our error messages.

Reply:
We normally put quotes around identifiers like `'${id}'`

Reply:
Just for clarity - this isn't @TanayParikh's message - it comes from existing code. But we can certainly take the opportunity to clean it up 😄

Reply:
Replacing with `JS object instance with Id '${id}' does not exist. It may have been disposed.`